src/yamlprocessor/tests/test_dataprocess.py

```diff
@@ -147,6 +147,47 @@ def test_process_variable_4():
             == '2022-02-20T22:02:00' + out3)
 
 
+def test_process_variable_5():
+    """Test DataProcessor.process_variable int, float, bool substitution."""
+    processor = DataProcessor()
+    processor.variable_map.clear()
+    # Variables for substitution
+    processor.variable_map.update({
+        'N_PLANETS': '8',
+        'COLD': '-10',
+        'PI': '3.14',
+        'CHARGE': '-1.6E-19',
+        'GOOD': 'true',
```

Comment:
Add 'BOOLTYPO': 'ttrue', and attempt to cast as bool (i.e. a cast of a string we want to fail when attempting to cast to a boolean)

Reply:
Is mixed case (True, False) supported as well?